tests/testthat/test-character_only_linter.R

```diff
@@ -0,0 +1,92 @@
+test_that("character_only_linter skips allowed usages", {
+  linter <- character_only_linter()
+
+  expect_lint("library(data.table)", NULL, linter)
+  expect_lint("function(pkg) library(pkg, character.only = TRUE)", NULL, linter)
+  expect_lint("function(pkgs) sapply(pkgs, require, character.only = TRUE)", NULL, linter)
```

Comment:
What makes this one different from the test in L 81?

Reply:
it's inside a `function()`. This usage is exempted by default to allow for packags like {knitr} which may load packages on the user's behalf.

Reply:
What about a similar bootstrapping code in "userland", like

```r
pkgs <- discover_dependencies()
sapply(pkgs, library, character.only = TRUE)
```
?

Do we want to block that?
(I'm fine with that since it's non-default, just asking)

Reply:
Not 100% sure what you mean by "userland" here, as in, at the top-level of a script?

Yes, that's the explicit purpose here -- discouraging usage like

```r
script_deps <- c("dplyr", "lubridate", "plm")
sapply(script_deps, library, character.only = TRUE)
```

It's much better for code analysis/tooling to have a consistent way of writing the script's dependencies, having usage like this makes it a mess to find out which script attaches which packages at scale.

Possibly we could add some parameters to fine-tune behavior here, but I'd leave that for when we get user feedback.

Reply:
Yeah, I was thinking about top-level scripts. I agree it should be blocked.